src/main/java/com/scalar/db/storage/dynamo/ScannerImpl.java

```diff
@@ -53,4 +58,70 @@ public ScannerImpl(
 
   @Override
   public void close() {}
+
+  private void sort() {
+    Scan scan = (Scan) selection;
+    DynamoOperation dynamoOperation = new DynamoOperation(scan, metadata);
+    if (dynamoOperation.isSingleClusteringKey()) {
+      // the ordering and the limitation already are applied in DynamoDB
+      return;
+    }
+
+    if (!scan.getOrderings().isEmpty()) {
+      Collections.sort(
```

Comment:
I think this part should be separated as another class like `ItemSorter` for separation of concerns and it should be tested independently to check the sorting capability intensively. (like testing all the types and complex sort orders because bugs can be easily introduced in this kind of code)

In that case, `ItemSorter` can be given in the constructor like this.
```
public ScannerImpl
    List<Map<String, AttributeValue>> items, Selection selection, TableMetadata metadata, ItemSorter sorter) {
```

Reply:
Thank you! I've added `ItemSorter`. Could you take a look again?